torch/nn/modules/module.py

```diff
@@ -663,16 +656,27 @@ def _load_from_state_dict(self, state_dict, prefix, metadata, strict, missing_ke
                                       'whose dimensions in the model are {} and '
                                       'whose dimensions in the checkpoint are {}.'
                                       .format(key, param.size(), input_param.size()))
-            elif strict:
-                missing_keys.append(key)
 
-        if strict:
-            for key, input_param in state_dict.items():
-                if key.startswith(prefix):
-                    input_name = key[len(prefix):]
-                    input_name = input_name.split('.', 1)[0]  # get the name of param/buffer/child
-                    if input_name not in self._modules and input_name not in local_state:
-                        unexpected_keys.append(key)
+    def diff_state_dict(self, state_dict):
+        r"""Compares current keys with given state dictionary
+        and returns matched, unmatched and unexpected keys
+
+        Arguments:
+            state_dict (dict): a dict containing parameters and
+                persistent buffers.
+        """
+        local_keys = set(self.state_dict())
+        state_keys, _ = zip(*state_dict.items())
+        state_keys = set(state_keys)
+
+        # matched keys are keys present in current and given state
+        matched_keys = state_keys & local_keys
+        # unmatched keys are present in current but not in given state
+        unmatched_keys = local_keys - state_keys
+        # unexpected keys are present in state but not in current
+        unexpected_keys = state_keys - local_keys
+
+        return list(matched_keys), list(unmatched_keys), list(unexpected_keys)
```

Comment:
I'm not 100% sure if we really want to have this function. The beginning could be simplified to
```py
local_keys = set(self.state_dict())
state_keys = set(state_dict)
```
and then it becomes a few lines of very straightforward Python code. To be honest it would be harder for me to find this function to the docs, and make sure that I assign the three lists we return in the correct order, than to write them out myself...